File: go/pickup_points.go
package zymeup

type PickupPointClient struct {
	client *Client
}

func NewPickupPointClient(c *Client) *PickupPointClient {
	return &PickupPointClient{client: c}
}

func (c *PickupPointClient) List(params map[string]interface{}) ([]map[string]interface{}, error) {
	path := "/api/v1/admin/pickup-points/" + buildQueryString(params)
	resp, err := c.client.doRequest("GET", path, nil)
	if err != nil {
		return nil, err
	}
	var result struct {
		Code int `json:"code"`
		Data struct {
			Items []map[string]interface{} `json:"items"`
			Total int                      `json:"total"`
		} `json:"data"`
	}
	if err := decodeResponse(resp, &result); err != nil {
		return nil, err
	}
	return result.Data.Items, nil
}

func (c *PickupPointClient) Get(id string) (map[string]interface{}, error) {
	resp, err := c.client.doRequest("GET", "/api/v1/admin/pickup-points/"+id, nil)
	if err != nil {
		return nil, err
	}
	var result struct {
		Code int                    `json:"code"`
		Data map[string]interface{} `json:"data"`
	}
	if err := decodeResponse(resp, &result); err != nil {
		return nil, err
	}
	return result.Data, nil
}

func (c *PickupPointClient) Create(data map[string]interface{}) (map[string]interface{}, error) {
	resp, err := c.client.doRequest("POST", "/api/v1/admin/pickup-points/", data)
	if err != nil {
		return nil, err
	}
	var result struct {
		Code int                    `json:"code"`
		Data map[string]interface{} `json:"data"`
	}
	if err := decodeResponse(resp, &result); err != nil {
		return nil, err
	}
	return result.Data, nil
}

func (c *PickupPointClient) Update(id string, data map[string]interface{}) (map[string]interface{}, error) {
	resp, err := c.client.doRequest("PUT", "/api/v1/admin/pickup-points/"+id, data)
	if err != nil {
		return nil, err
	}
	var result struct {
		Code int                    `json:"code"`
		Data map[string]interface{} `json:"data"`
	}
	if err := decodeResponse(resp, &result); err != nil {
		return nil, err
	}
	return result.Data, nil
}

func (c *PickupPointClient) Deactivate(id string) error {
	resp, err := c.client.doRequest("POST", "/api/v1/admin/pickup-points/"+id+"/deactivate", nil)
	if err != nil {
		return err
	}
	return decodeResponse(resp, nil)
}

func (c *PickupPointClient) Search(params map[string]interface{}) ([]map[string]interface{}, error) {
	path := "/api/v1/admin/pickup-points/" + buildQueryString(params)
	resp, err := c.client.doRequest("GET", path, nil)
	if err != nil {
		return nil, err
	}
	var result struct {
		Code int `json:"code"`
		Data struct {
			Items []map[string]interface{} `json:"items"`
			Total int                      `json:"total"`
		} `json:"data"`
	}
	if err := decodeResponse(resp, &result); err != nil {
		return nil, err
	}
	return result.Data.Items, nil
}

File: rust/src/pickup_points.rs
use reqwest::Method;

use crate::error::Result;
use crate::http_client::HttpClient;
use crate::types::{ListResponse, PickupPoint};

pub struct PickupPointClient {
    pub(crate) inner: HttpClient,
}

impl PickupPointClient {
    pub fn new(inner: HttpClient) -> Self {
        Self { inner }
    }
}

impl PickupPointClient {
    pub async fn list(
        &self,
        active_only: Option<bool>,
    ) -> Result<ListResponse<PickupPoint>> {
        let mut path = "/api/v1/admin/pickup-points/".to_string();
        if let Some(ao) = active_only {
            path.push_str(&format!("?active_only={}", ao));
        }
        self.inner.request(&path, Method::GET, None).await
    }

    pub async fn get(&self, id: &str) -> Result<PickupPoint> {
        self.inner
            .request(
                &format!("/api/v1/admin/pickup-points/{}", id),
                Method::GET,
                None,
            )
            .await
    }

    pub async fn create(
        &self,
        body: serde_json::Value,
    ) -> Result<PickupPoint> {
        self.inner
            .request(
                "/api/v1/admin/pickup-points/",
                Method::POST,
                Some(body),
            )
            .await
    }

    pub async fn update(
        &self,
        id: &str,
        body: serde_json::Value,
    ) -> Result<PickupPoint> {
        self.inner
            .request(
                &format!("/api/v1/admin/pickup-points/{}", id),
                Method::PUT,
                Some(body),
            )
            .await
    }

    pub async fn deactivate(&self, id: &str) -> Result<serde_json::Value> {
        self.inner
            .request(
                &format!("/api/v1/admin/pickup-points/{}/deactivate", id),
                Method::POST,
                Some(serde_json::json!({})),
            )
            .await
    }
}
